util/clock/time_test.go

```diff
@@ -37,8 +37,12 @@ var (
 )
 
 func TestSleepContext(t *testing.T) {
+	t.Parallel()
+
 	for _, tc := range cases {
 		t.Run(fmt.Sprintf("%v:%v", tc.dur, tc.timeout), func(t *testing.T) {
+			t.Parallel()
+
 			ctx, cancel := context.WithTimeout(context.Background(), tc.timeout)
```

Comment:
I think this will fall foul of https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables.